data/importexport/document_downloader.py

```diff
@@ -0,0 +1,66 @@
+import os
+from time import sleep
+from qgis.core import QgsNetworkAccessManager
+from PyQt4.QtNetwork import (
+    QNetworkAccessManager,
+    QNetworkRequest
+)
+from PyQt4.QtGui import (
+    QApplication,
+)
+from PyQt4.QtCore import (
+    QUrl,
+    QByteArray,
+    QEventLoop,
+)
+
+class Doc_Downloader(object):
+    """
+    this class 
+    """
+    def __init__(self, username, password):
+        #self.username = username
+        #self.password = password
+        self.manager = QNetworkAccessManager()
+        self.manager.finished.connect(self.handle_download)
+        self.header_data = QByteArray(
+            '{}:{}'.format(username, password)
+        ).toBase64()
+
+    def kobo_download(self, src_url, dest_filename):
+        self.download_result = None
+        self.dest_filename = dest_filename
+        request = QNetworkRequest(QUrl(src_url))
+        request.setRawHeader(
+            'Authorization',
+            'Basic {}'.format(self.header_data)
+        )
+        try:
+            self.manager.get(request)
+            self.loop = QEventLoop()
+            self.loop.exec_()
+        except:
+            pass
```

Comment:
Ideally, the exception handler should catch and report any error thrown; '**pass**' will hide any exception that might be thrown. 
Since the get request will report any errors in the request handler, it seems there is no need for the try-except.

Reply:
Done @pgathogo